infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/LogicalGroupBasedLogEntryReader.java

```diff
@@ -0,0 +1,28 @@
+package org.corfudb.infrastructure.logreplication.replication.send.logreader;
+
+import org.corfudb.infrastructure.logreplication.LogicalGroupBasedReplicationConfig;
+import org.corfudb.infrastructure.logreplication.infrastructure.ReplicationSession;
+import org.corfudb.infrastructure.logreplication.utils.LogReplicationConfigManager;
+import org.corfudb.runtime.CorfuRuntime;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class LogicalGroupBasedLogEntryReader extends BaseLogEntryReader {
+
+    public LogicalGroupBasedLogEntryReader(CorfuRuntime runtime, LogReplicationConfigManager configManager,
+                                                 ReplicationSession replicationSession) {
+        super(runtime, configManager, replicationSession);
+    }
+
+    @Override
+    protected void refreshStreamsToReplicateSet() {
```

Comment:
Why are we overriding this function? The overridden implementations in the children are the same..so why not have them in the parent class itself?

Reply:
Agreed, I think this class isn't complete anyway, let me move it out. thanks